### DISCUSSION TITLE
How to show specific column on a specific tab

### DISCUSSION THREAD

**Discussion by naseef** [filamentphp/filament#9572] [Help] [answered]
In the list record `getTabs()` , how can I hide some columns based on the selected tab?

for example, I have 3 columns( `name`, `type` and `designation` )and 2 tabs, (`all` and `active`). I want `type` column to be hidden in the `active` tab.  How to achieve this?

**Comment by leandrocfe** [comment, marked as answer]
```php
TextColumn::make('type')
->hidden(fn ($livewire) => $livewire->activeTab === 'active')
```

**Reply by naseef** [reply]
Thanks

**Reply by taher-k5** [reply]
Thanks I am finding this for days

**Reply by mkwsra** [reply]
It's very odd, on first click on the tab, it's not working, clicking again on the same tab, the logic is working fine. What am I missing here?

**Reply by anas43950** [reply]
@leandrocfe brother with filament 5, I am getting error: Property [$activeTab] not found on component when tryint to use your solution

**Reply by thaqebon** [reply]
> It's very odd, on first click on the tab, it's not working, clicking again on the same tab, the logic is working fine. What am I missing here?


I guess the issue is `bootedInteractsWithTable()` runs before `aciveTab` been updated.
Overwrite `updatedActiveTab()` may fix the issue. 
```php
    public function updatedActiveTab(): void
    {
        $this->traitBootedInteractsWithTable();
        $this->traitUpdatedActiveTab();
    }
```
